### PR TITLE
Activate virtual environment within pulse.py (#343)

### PR DESCRIPTION
**Pull request by whimboo**
This will fix issue #343, and which bugs me for a while now. So I kinda would like to see it landed.

### CODE REVIEW

**Comment by whimboo** [comment]
@sydvicious and/or @mjzffr could at least one of you have a look? Thanks.

File: lib/automation.py [hunk 1]
```diff
@@ -0,0 +1,260 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import ConfigParser
```
Comment (whimboo) [inline lib/automation.py]: I should have made this a separate commit. It's just a plain 1:1 copy of the FirefoxAutomation class.

**Comment by whimboo** [comment]
Lets wait with the review. I will also sync up start.py.

**Comment by whimboo** [comment]
Ok, latest code is up and will hard-require an existing virtualenv now. It's ready to be reviewed.

File: start.py [hunk 2]
```diff
@@ -40,17 +41,15 @@ def download_jenkins():
...
-        # for more info see:
-        # http://www.virtualenv.org/en/latest/#using-virtualenv-without-bin-python
         execfile(JENKINS_ENV, dict(__file__=JENKINS_ENV))
         print "Virtual environment activated successfully."
-    except IOError:
-        print "Could not activate virtual environment."
-        print "Exiting."
+    except Exception as ex:
+        print 'Could not activate virtual environment at "%s": %s.' % (JENKINS_ENV, str(ex))
         sys.exit(IOError)
```
Comment (mjzffr) [inline start.py]: Why exit with IOError when you're catching Exception?

**Comment by mjzffr** [comment]
One question. 
LGTM overall.

File: start.py (hunk 2 above)
Comment (whimboo) [inline start.py]: That existed already before. So we could re-throw the exception or still call `exit()`. For the latter we could e.g. exit with 1.

**Comment by whimboo** [comment]
@mjzffr I pushed an update. Are you happy with it?

**Comment by mjzffr** [comment]
@whimboo Yep! :pineapple:

**Pull request closed without merging**